avocado/utils/ssh.py

```diff
@@ -214,3 +214,28 @@ def quit(self):
         :rtype: bool
         """
         return self._master_command('exit')
+
+    def copy_files(self, source, destination, recursive=False):
+        """
+        Copy Files to and from remote through scp session
+        :param source: Local file
+        :type: str
+        :param destination: Remote file location
+        :type: str
+        :param recursive: Scp option for copy file. if set to True
+                          copy files inside directory recursively.
+        :type: boot
+        :returns: Whether the files copy successfully or not.
+        :rtype: bool
+        """
+        cmd = path_utils.find_command('scp')
+        options = self._dash_o_opts_to_str(self.DEFAULT_OPTIONS)
+        if recursive:
+            options += ' -r'
+        options += " {} {}".format(source, destination)
+        try:
+            result = process.run("{} {}".format(cmd, options),
+                                 ignore_status=True)
+            return result.exit_status == 0
+        except process.CmdError as exc:
+            raise ("failed to copy file {}".format(exc))
```

Comment:
Did you execute this code? Because looks like this line have a syntax problem:

```
TypeError: exceptions must be old-style classes or derived from BaseException, not str
```

Reply:
yes tested the code, i not getting any syntax error.

Reply:
```python3
#!/bin/env python3
try:
    a = 10 / 0
except ZeroDivisionError as exc:
    raise ("failed to copy file {}".format(exc))
```
```bash
./test.py
Traceback (most recent call last):
  File "/tmp/test2.py", line 3, in <module>
    a = 10 / 0
ZeroDivisionError: division by zero

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/test2.py", line 5, in <module>
    raise ("failed to copy file {}".format(exc))
TypeError: exceptions must derive from BaseException
```

This code it is causing **another** exception on top of the original one. Maybe you need to raise a proper exception:

```python
raise MyCustomException("Failed to copy file {}".format(exc))
```